utilities/list-helpers.metta

```diff
@@ -260,4 +260,29 @@
 (= (foldl $func (Cons $x $xs))
     (foldLeft $func $x $xs)
 )
->>>>>>> df941223a71da2045e09dbc32103c58db2d57f76
+
+ ;Function to remove an element at a given index in a List.
+(: removeAtIndex (-> Number (List $t) (List $t)))
+(= (removeAtIndex $a Nil) Nil)
+
+(= (removeAtIndex $a (Cons $x $xs))
+    (if (== $a 0)
+        (removeAtIndex (- $a 1) $xs)
+        (Cons $x (removeAtIndex (- $a 1) $xs))
+    )
+)
+
+ ;A Function to replace an element at a given index in a List.
+(: replaceAtIndex (-> Number $element (List $t) (List $t)))
```

Comment:
the type of the element to should be the same as the types of the elements found in the list